crypto/engine/eng_openssl.c

```diff
@@ -513,22 +519,28 @@ static int ossl_hmac_copy(EVP_PKEY_CTX *dst, EVP_PKEY_CTX *src)
     dctx = EVP_PKEY_CTX_get_data(dst);
     dctx->md = sctx->md;
     if (!HMAC_CTX_copy(dctx->ctx, sctx->ctx))
-        return 0;
+        goto err;
     if (sctx->ktmp.data) {
         if (!ASN1_OCTET_STRING_set(&dctx->ktmp,
                                    sctx->ktmp.data, sctx->ktmp.length))
-            return 0;
+            goto err;
     }
     return 1;
+err:
+    ossl_hmac_cleanup(dst);
```

Comment:
Since this code is undocumented, I have no idea if this is the right thing to do or not.  EVP_PKEY_CTX_get_data(dst) will return a dagling pointer now, which isn't good in any case.  ossl_hmac_cleanup() should probably set it to NULL.

We call ossl_hmac_init() here, which means that if already existed we leaked the old value.  And I'm not sure how dst was created.